static/webpack-config.js

```diff
@@ -25,11 +26,29 @@ module.exports = function () {
     });
   }
 
+  const globalConfigPath = `./${jamboConfig.dirs.config}/global_config.json`;
+  let globalConfig = null;
+  if (fs.existsSync(globalConfigPath)) {
+    globalConfigRaw = fs.readFileSync(globalConfigPath, 'utf-8');
+    globalConfig = parse(globalConfigRaw);
+  }
+
+  const useJWT = globalConfig && globalConfig.useJWT
+  let jamboInjectedData = process.env.JAMBO_INJECTED_DATA || null;
+  jamboInjectedData = jamboInjectedData && JSON.parse(jamboInjectedData);
+
+  const getSecuredJamboInjectedData =
+    require(`./${jamboConfig.dirs.output}/static/webpack/getSecuredJamboInjectedData.js`);
+
+  const updatedJamboInjectedData = useJWT 
+    ? getSecuredJamboInjectedData(jamboInjectedData)
+    : jamboInjectedData
+
   const plugins = [
     new MiniCssExtractPlugin({ filename: '[name].css' }),
     ...htmlPlugins,
-    new webpack.EnvironmentPlugin({
-      JAMBO_INJECTED_DATA: null
+    new webpack.DefinePlugin({
+      'process.env.JAMBO_INJECTED_DATA': JSON.stringify(JSON.stringify(updatedJamboInjectedData))
```

Comment:
For my own edification, what's the scope of using the `JAMBO_INJECTED_DATA` in our Webpack toolchain? I believe we use it in the iFrame assets to get domains and such. Are we embedding the whole object in any of our static assets or using the apiKey in any of the static assets?

Reply:
We do put the whole object in a few spots, and that ends up in many of our assets.

Webpack puts it into the HitchhikerJS.getInjectedProp() function. The only spot where that accesses the apiKey in our code is in core.hbs, but I disabled that access when useJWT is true. It's possible that someone might be trying to get the apiKey from getInjectedProp(), but that really doesn't make sense if a site is built with useJWT=true.

Secondly, it's put in the InjectedData class which simply provides info about the domains, so the apiKey is not needed there.